src/js/auth.js

```diff
@@ -0,0 +1,65 @@
+const { nanoid } = 'nanoid';
+
+const auth = () => {
+  const getSession = () => {
+    const sessionData = sessionStorage.getItem('moshoodabdullahi-todo-list-app-session');
+    return sessionData ? JSON.parse(sessionData) : null;
+  };
+
+  const getLocalStorageUser = (userEmail) => {
+    const userArray = JSON.parse(localStorage.getItem('userArray')) || [];
+    const foundUser = userArray.find((userData) => userData.email === userEmail);
+
+    if (foundUser) {
+      return foundUser.session;
+    }
+    return null;
+  };
+
+  const addLocalStorageUser = (sessionData) => {
+    const userArray = JSON.parse(localStorage.getItem('userArray')) || [];
+    userArray.push(sessionData);
+
+    localStorage.setItem('userArray', JSON.stringify(userArray));
+  };
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const signIn = (email) => {
+    if (!validateEmail(email)) {
+      throw new Error('Invalid email format');
+    }
+
+    const user = getLocalStorageUser(email);
+
+    if (user) {
+      sessionStorage.setItem('moshoodabdullahi-todo-user', JSON.stringify(user));
+      return sessionStorage.getItem('user');
```

Comment:
What is this for? How any session do you have? I am seeing three session with different keys already. We should use only one session, right?

Reply:
Done